nncf/quantization/algorithms/accuracy_control/ranker.py

```diff
@@ -290,10 +291,14 @@ def _collect_values_for_each_item(self, model: TModel, data_items: Iterable[Any]
         :param data_items: Data items.
         :return: A list that contains a metric for each item from the dataset.
         """
-        metrics = [
-            self._validation_fn(self._algo_backend.prepare_for_inference(model), [data_item]) \
-                for data_item in data_items
-        ]
+        metrics = []
+        # pylint: disable=protected-access
+        indexed_items = zip(data_items._indices, data_items) if data_items._indices else enumerate(data_items)
```

Comment:
Could you briefly explain these changes? It looks very ugly.

Reply:
Not applicable now. The code was changed.